File: sdkwork-backend-sdk-swift/Sources/API/SchemaApi.swift
import Foundation

public class SchemaApi {
    private let client: HttpClient
    
    public init(client: HttpClient) {
        self.client = client
    }

    /// Update an existing database schema
    public func update(body: PlusSchemaForm) async throws -> PlusApiResultPlusSchemaVO? {
        let response = try await client.put(ApiPaths.backendPath("/schema"), body: body)
        return response as? PlusApiResultPlusSchemaVO
    }

    /// Create a new database schema
    public func create(body: PlusSchemaForm) async throws -> PlusApiResultPlusSchemaVO? {
        let response = try await client.post(ApiPaths.backendPath("/schema"), body: body)
        return response as? PlusApiResultPlusSchemaVO
    }

    /// Get database schemas by page
    public func listByPage(body: QueryListForm? = nil, params: [String: Any]? = nil) async throws -> PlusApiResultPagePlusSchemaVO? {
        let response = try await client.post(ApiPaths.backendPath("/schema/list"), body: body, params: params)
        return response as? PlusApiResultPagePlusSchemaVO
    }

    /// Get all database schemas
    public func listAllEntities(body: QueryListForm? = nil) async throws -> PlusApiResultListPlusSchemaVO? {
        let response = try await client.post(ApiPaths.backendPath("/schema/list/all"), body: body)
        return response as? PlusApiResultListPlusSchemaVO
    }

    /// Get a database schema by ID
    public func getById(id: String) async throws -> PlusApiResultPlusSchemaVO? {
        let response = try await client.get(ApiPaths.backendPath("/schema/\(id)"))
        return response as? PlusApiResultPlusSchemaVO
    }

    /// Delete a database schema
    public func delete(id: String) async throws -> PlusApiResultBoolean? {
        let response = try await client.delete(ApiPaths.backendPath("/schema/\(id)"))
        return response as? PlusApiResultBoolean
    }
}

File: sdkwork-backend-sdk-csharp/Api/SchemaApi.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Backend.Http;
using Backend.Models;

namespace Backend.Api
{
    public class SchemaApi
    {
        private readonly HttpClient _client;

        public SchemaApi(HttpClient client)
        {
            _client = client;
        }

        /// <summary>
        /// Update an existing database schema
        /// </summary>
        public async Task<PlusApiResultPlusSchemaVO?> UpdateAsync(PlusSchemaForm body)
        {
            return await _client.PutAsync<PlusApiResultPlusSchemaVO>(ApiPaths.BackendPath("/schema"), body);
        }

        /// <summary>
        /// Create a new database schema
        /// </summary>
        public async Task<PlusApiResultPlusSchemaVO?> CreateAsync(PlusSchemaForm body)
        {
            return await _client.PostAsync<PlusApiResultPlusSchemaVO>(ApiPaths.BackendPath("/schema"), body);
        }

        /// <summary>
        /// Get database schemas by page
        /// </summary>
        public async Task<PlusApiResultPagePlusSchemaVO?> ListByPageAsync(QueryListForm? body = null, Dictionary<string, object>? query = null)
        {
            return await _client.PostAsync<PlusApiResultPagePlusSchemaVO>(ApiPaths.BackendPath("/schema/list"), body, query);
        }

        /// <summary>
        /// Get all database schemas
        /// </summary>
        public async Task<PlusApiResultListPlusSchemaVO?> ListAllEntitiesAsync(QueryListForm? body = null)
        {
            return await _client.PostAsync<PlusApiResultListPlusSchemaVO>(ApiPaths.BackendPath("/schema/list/all"), body);
        }

        /// <summary>
        /// Get a database schema by ID
        /// </summary>
        public async Task<PlusApiResultPlusSchemaVO?> GetByIdAsync(string id)
        {
            return await _client.GetAsync<PlusApiResultPlusSchemaVO>(ApiPaths.BackendPath($"/schema/{id}"));
        }

        /// <summary>
        /// Delete a database schema
        /// </summary>
        public async Task<PlusApiResultBoolean?> DeleteAsync(string id)
        {
            return await _client.DeleteAsync<PlusApiResultBoolean>(ApiPaths.BackendPath($"/schema/{id}"));
        }
    }
}
